### PR TITLE
[NPU] enhance accuracy for model kimi-vl-a3b-instruct

### PR DESCRIPTION
**Pull request by McZyWu**
Co-authored-by: randgun@github.com hexq0210@github.com

## Motivation
previously accuracy for kimi-vl-a3b-instruct is no more than 0.05.

The Kimi-VL-A3B-Instruct model exhibited severe accuracy issues in the current codebase, with benchmark accuracy no higher than 0.05 (5%) —far below the expected performance baseline. Root cause analysis identified misalignment with prerequisite implementations and missing critical KV cache handling, which rendered the model effectively non-functional for practical use.

## Modifications
as follows

The changes made to the 4 documents are to align them with the prior prerequisite PR. https://github.com/sgl-project/sglang/pull/17007. being coherent with npu version: https://github.com/sgl-project/sglang/pull/16436.
Besides, the changes of forward_batch.token_to_kv_pool.set_kv_buffer is to save kv cache. the accuracy would be 0 without this.
the change of [python/sglang/srt/hardware_backend/npu/moe/topk.py] is to be consistent with the documentation of operator  npu_moe_gating_top_k, adapting to the operator input.

## Accuracy Tests
After applying the above changes, the model’s accuracy saw a dramatic improvement across all critical execution modes:
Eagle Mode: Accuracy reaches at least 0.67 (67%)
Graph Mode: Accuracy reaches at least 0.67 (67%)

image
image

P.S. prerequisite dependency: custom-ops

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py [hunk 1]
```diff
@@ -102,6 +102,9 @@ def forward_mha_prepare_npu(
         k_pe = latent_cache[:, :, m.kv_lora_rank :]
         if m.rotary_emb is not None:
             q_pe, k_pe = m.rotary_emb(positions, q_pe, k_pe)
+        forward_batch.token_to_kv_pool.set_kv_buffer(
```
Comment (iforgetmyname) [inline python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py:107]: add comments

File: python/sglang/srt/hardware_backend/npu/moe/topk.py [hunk 2]
```diff
@@ -38,7 +38,11 @@ def fused_topk_npu(
             )
         topk_weights = topk_weights.to(torch.float32)
 
-    elif use_grouped_topk and correction_bias is not None:
+    elif (
+        use_grouped_topk
+        and correction_bias is not None
+        and router_logits.shape[-1] <= 2048
+    ):
```
Comment (iforgetmyname) [inline python/sglang/srt/hardware_backend/npu/moe/topk.py]: add comments on `router_logits.shape[-1] <= 2048`

File: python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py (hunk 1 above)
Comment (McZyWu) [reply python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py:107]: Thank you!

File: python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py [hunk 3]
```diff
@@ -103,6 +103,9 @@ def forward_mha_prepare_npu(
...
+            m, forward_batch.out_cache_loc, kv_a.unsqueeze(1), k_pe
+        )  # this is for model kimi-vl-a3B-instruct
```
Comment (ping1jing2) [inline python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py]: it would be better to do it like below
```python
# this is for model kimi-vl-a3B-instruct
forward_batch.token_to_kv_pool.set_kv_buffer(
```

**Review by ping1jing2 — CHANGES_REQUESTED** [review]

File: python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py (hunk 3 above)
Comment (McZyWu) [reply python/sglang/srt/hardware_backend/npu/modules/deepseek_v2_attention_mla_npu.py]: sure thanks

**Review by ping1jing2 — APPROVED** [review]

**Comment by iforgetmyname** [comment]
this pr should not affect another platforms, if casuing any issue, just revert it
